### PR TITLE
Add execution blockNumber to the notifier log

### PR DESCRIPTION
**Pull request by g11tech**
Add execution blockNumber to the notifier log

We only display execution head's blockHash while execution block number is also an important info that people from EL land (ethereumjs specifically) have asked for in the log.

result:
image

### CODE REVIEW

File: packages/api/src/beacon/routes/debug.ts [hunk 1]
```diff
@@ -29,6 +29,7 @@ const stringType = new StringType();
 const protoNodeSszType = new ContainerType(
   {
     executionPayloadBlockHash: stringType,
+    executionPayloadNumber: ssz.UintNum64,
```
Comment (dapplion) [inline packages/api/src/beacon/routes/debug.ts:32]: Is this API standardized?

**Review by dapplion — COMMENTED** [review]
Not strongly opposed to this but the status log is already pretty long. Maybe we should add some guidelines so the status log does not grow forever. i.e. let's try to formalize a bit the purpose of this log and what information makes it in, and what does not. I can imagine that with new forks more and more data would be appealing to be displayed here

File: packages/api/src/beacon/routes/debug.ts (hunk 1 above)
Comment (g11tech) [reply packages/api/src/beacon/routes/debug.ts:32]: ummm ... i don't think so, double checked

**Comment by g11tech** [comment]
> Not strongly opposed to this but the status log is already pretty long. Maybe we should add some guidelines so the status log does not grow forever. i.e. let's try to formalize a bit the purpose of this log and what information makes it in, and what does not. I can imagine that with new forks more and more data would be appealing to be displayed here

right!, may be we can further truncate the hash of head exec-block since number + first 4 hash bytes are good enough to identify where execution head is:  so like `exec-block: valid(16885600 0x2013…)`
the log would look like:
 
`Mar-22 20:47:17.001[]                 info: Synced - slot: 6057834 - head: 6057834 0xff7e…d00c exec-block: valid(16885600 0x2013…) - finalized: 0xce5a…bdac:189305 (exec-block: 16885526) - peers: 17`

We can also apply same principal for beacon head so: ` head: 6057834 0xff7e… exec-block: valid(16885600 0x2013…) `

May be we can also drop exec-block bit from finalized and let it remain unchanged, so final log could look like:
`Mar-22 20:47:17.001[]                 info: Synced - slot: 6057834 - head: 6057834 0xff7e… exec-block: valid(16885600 0x2013…) - finalized: 0xce5a…bdac:189305 - peers: 17`

**Comment by dapplion** [comment]
Yes definitely better not to add in the finalized, so this log looks good

> May be we can also drop exec-block bit from finalized and let it remain unchanged, so final log could look like:
Mar-22 20:47:17.001[]                 info: Synced - slot: 6057834 - head: 6057834 0xff7e… exec-block: valid(16885600 0x2013…) - finalized: 0xce5a…bdac:189305 - peers: 17

**Comment by g11tech** [comment]
> Yes definitely better not to add in the finalized, so this log looks good
> 
> > May be we can also drop exec-block bit from finalized and let it remain unchanged, so final log could look like:
> > Mar-22 20:47:17.001[]                 info: Synced - slot: 6057834 - head: 6057834 0xff7e… exec-block: valid(16885600 0x2013…) - finalized: 0xce5a…bdac:189305 - peers: 17

how about this:
```typescript
Mar-24 11:09:05.434[]                 info: Syncing - 10 minutes left - 6.32 slots/s - head: 6065535 (clock: 6069343) 0x6c82…486d exec-block: valid(16893200 0xadfe…) - finalized: 0x14bd…d8d6:189545 - peers: 7
Mar-24 11:09:17.473[]                 info: Syncing - 9.8 minutes left - 6.35 slots/s - head: 6065599 (clock: 6069344) 0xb659…7109 exec-block: valid(16893264 0xf32f…) - finalized: 0xb551…b203:189548 - peers: 9
Mar-24 11:16:29.187[]                 info: Syncing - 3.4 minutes left - 4.77 slots/s - head: 6068415 (clock-965) 0xeddf…e97b exec-block: valid(16896048 0x5d16…) - finalized: 0xd7ba…8386:189636 - peers: 9
valid(16896681 0xc9a6…9b3a) - finalized: 0x495d…f884:189656 - peers: 10
Mar-24 11:18:53.033[]                 info: Syncing - 42 seconds left - 5.77 slots/s - head: 6069151 (clock-241) 0xdb52…77b7 exec-block: valid(16896777 0x0c24…) - finalized: 0x6777…d015:189659 - peers: 17
Mar-24 11:19:17.031[]                 info: Syncing - 18 seconds left - 6.55 slots/s - head: 6069279 (clock-115) 0xd6bb…4ecd exec-block: valid(16896905 0x6599…) - finalized: 0x4658…51b6:189663 - peers: 11
Mar-24 11:19:29.047[]                 info: Syncing - 7.8 seconds left - 6.67 slots/s - head: 6069343 (clock-52) 0xf111…4552 exec-block: valid(16896968 0x6a16…) - finalized: 0x4e1e…3499:189665 - peers: 14
Mar-24 11:19:35.675[network]          info: Subscribed gossip core topics
Mar-24 11:37:29.001[]                 info: Synced - head: 6069482 0xa9fd…ddab exec-block: valid(16897109 0x4951…) - finalized: 0xec9e…9cc1:189669 - peers: 27
Mar-24 11:37:17.000[]                 info: Synced - head: 6069482 (clock-1) 0x88b2…fb8c exec-block: valid(16897108 0x90c9…) - finalized: 0xec9e…9cc1:189669 - peers: 31
Mar-24 11:37:29.001[]                 info: Synced - head: 6069484 0xa9fd…ddab exec-block: valid(16897109 0x4951…) - finalized: 0xec9e…9cc1:189669 - peers: 27
```

Explanation:
explanation : 
1. if clock -head > 1000: 
`head: 6065535 (clock: 6069343) 0x6c82…486d exec-block: valid(16896873 0x029b…)`
2. if clock - head  < 1000:
`head: 6069247 (clock - 146) 0x824f…7bbd exec-block: valid(16896873 0x029b…)`
3. if clock === head
`head: 6069485 0xa9fd…ddab exec-block: valid(16897109 0x4951…)`
4. again if skipped slot
`head: 6069482 (clock - 1) 0x88b2…fb8c exec-block: valid(16897108 0x90c9…)`

**Comment by philknows** [comment]
I think it no matter what the format is, we should include in our docs what each thing in the log means. I was used to the `skipped 1` previously which made sense instantly, but `clock-1` is fine too as long as we explain somewhere that means 1 skipped block (usually from missed proposals).

**Comment by g11tech** [comment]
> I think it no matter what the format is, we should include in our docs what each thing in the log means. I was used to the `skipped 1` previously which made sense instantly, but `clock-1` is fine too as long as we explain somewhere that means 1 skipped block (usually from missed proposals).

added

**Review by dapplion — DISMISSED** [review]
Disabled auto-merging to allow others to :+1: before merging

**Comment by philknows** [comment]
I had a chat with @wemeetagain yesterday about this PR, specifically about the logging which can be further shortened in our opinion to reduce any redundancy of information. If we can find the most concise way to explain something that is very clear without additional context and definitions in docs, we've hit the jackpot. 

At least for the slot numbers and identifying where the head is, it could be shortened to:

```
// synced and the head is at the latest slot. Reduces repeating the slot number twice if at the head.
Synced - slot: N - head: 0xabcd...dcba - ...

// synced and the head is one slot behind. We think this makes more sense comparatively to `clock-x` which was more confusing than `skipped 1`
Synced - slot: M - head: (slot -1) 0xabcd...dcba - ...
```

**Comment by g11tech** [comment]
~hmmm this looks good as well, will update~

Problem with slot as clock in the front is that when syncing from far away, this is confusing to others:
`Syncing .... - slot: 6069482 - head: 6065535 ...` 

as slot and head are both very generic , also clock slot info as the first thing isn't the concern of the users:

What is relevant to them are:

1. Am i synced or syncing
2. Where my local head is and if i am missing any latest blocks 

current clock info is only relevant w.r.t. the local head

**Comment by philknows** [comment]
> What is relevant to them are:
> 1. Am i synced or syncing
> 2. Where my local head is and if i am missing any latest blocks

So for syncing, you're right and that a lot of this information may not even be required. Maybe we can rethink something for `Syncing` 

We currently see something like:
```ts
info: Syncing - 2.8 hours left - 3.19 slots/s - slot: 5314216 (skipped 32552) - head: 5281664 0x14d0…8776
info: Syncing - 2.8 hours left - 3.22 slots/s - slot: 5314217 (skipped 32490) - head: 5281727 0x1d24…d906
info: Syncing - 2.7 hours left - 3.32 slots/s - slot: 5314218 (skipped 32459) - head: 5281759 0x0646…8590
``` 

And with the above idea we will see:
```ts
info: Syncing - 2.8 hours left - 3.19 slots/s - slot: 5314216 - head: (slot -32552) 0x14d0…8776
info: Syncing - 2.8 hours left - 3.22 slots/s - slot: 5314217 - head: (slot -32490) 0x1d24…d906
info: Syncing - 2.7 hours left - 3.32 slots/s - slot: 5314218 - head: (slot -32459) 0x0646…8590
```

1. As the user, we know that the node is still syncing, so much of this information is not really even necessary. Users know it's syncing. But now, how do we answer Q2?

2. Perhaps while syncing _only_, the easiest way to solve this is to have something different that shows where we are locally instead of where the chain is currently at. An idea here is just to omit the clock/slot.
```ts
info: Syncing - 2.8 hours left - 3.19 slots/s - head: 5281664 0x14d0…8776
info: Syncing - 2.8 hours left - 3.22 slots/s - head: 5281727 0x1d24…d906
info: Syncing - 2.7 hours left - 3.32 slots/s - head: 5281759 0x0646…8590
```

**Comment by g11tech** [comment]
Ok this is how the sync logs will look r.n.
I guess we should just merge this and relook at it separately as the main aim of the PR was to add exec block number/hash info of head 

```
Apr-20 15:24:08.034[]                 info: Searching peers - peers: 0 - slot: 6265018 - head: 6264018 0xed93…7b0a - exec-block: syncing(17088476 0x9649…) - finalized: 0xbf30…7e7c:195777
Apr-20 15:24:17.000[]                 info: Searching peers - peers: 0 - slot: 6265019 - head: 6264018 0xed93…7b0a - exec-block: syncing(17088476 0x9649…) - finalized: 0xbf30…7e7c:195777

Apr-20 15:13:41.298[]                 info: Syncing - 2.5 minutes left - 2.78 slots/s - slot: 6264966 - head: 6262966 0x5cec…f5b8 - exec-block: valid(17088105 0x6f74…) - finalized: 0x5cc0…3874:195764 - peers: 1
Apr-20 15:13:41.298[]                 info: Syncing - 2 minutes left - 2.78 slots/s - slot: 6264967 - head: 6263965 0x5cec…f5b8 - exec-block: valid(17088105 0x6f74…) - finalized: 0x5cc0…3874:195764 - peers: 1

Apr-20 15:13:53.151[]                 info: Syncing - 1.6 minutes left - 3.82 slots/s - slot: 6264967 - head: (slot -360) 0xe0cf…9f3c - exec-block: valid(17088167 0x2d6a…) - finalized: 0x8f3f…2f81:195766 - peers: 5
Apr-20 15:14:05.425[]                 info: Syncing - 1.1 minutes left - 4.33 slots/s - slot: 6264968 - head: (slot -297) 0x3655…1658 - exec-block: valid(17088231 0xdafd…) - finalized: 0x9475…425a:195769 - peers: 2
Apr-20 15:14:53.001[]                 info: Syncing - 9 seconds left - 5.00 slots/s - slot: 6264972 - head: (slot -45) 0x44e4…20a4 - exec-block: valid(17088475 0xca61…) - finalized: 0x9cbd…ba83:195776 - peers: 8

Apr-20 15:15:01.443[network]          info: Subscribed gossip core topics
Apr-20 15:15:01.446[sync]             info: Subscribed gossip core topics
Apr-20 15:15:05.000[]                 info: Synced - slot: 6264973 - head: 0x90ea…c655 - exec-block: valid(17088521 0xca9b…) - finalized: 0x6981…682f:195778 - peers: 6
Apr-20 15:15:17.003[]                 info: Synced - slot: 6264974 - head: 0x4f7e…0e3a - exec-block: valid(17088522 0x08b1…) - finalized: 0x6981…682f:195778 - peers: 6

Apr-20 15:15:41.001[]                 info: Synced - slot: 6264976 - head: (slot -1) 0x17c6…71a7 - exec-block: valid(17088524 0x5bc1…) - finalized: 0x6981…682f:195778 - peers: 8
Apr-20 15:15:53.001[]                 info: Synced - slot: 6264977 - head: (slot -2) 0x17c6…71a7 - exec-block: valid(17088524 0x5bc1…) - finalized: 0x6981…682f:195778 - peers: 8

Apr-20 15:16:05.000[]                 info: Synced - slot: 6264978 - head: 0xc9fd…28c5 - exec-block: valid(17088526 0xb5bf…) - finalized: 0x6981…682f:195778 - peers: 8
Apr-20 15:16:17.017[]                 info: Synced - slot: 6264979 - head: 0xde91…d4cb - exec-block: valid(17088527 0xa488…) - finalized: 0x6981…682f:195778 - peers: 7
```

**Review by wemeetagain — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.8.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.8.0) :tada:

**Pull request merged**